content/articles/servlet-request-and-servlet-collaboration/SevletRequest and ServletCallaboration.md

```diff
@@ -0,0 +1,277 @@
+- Some basic knowledge of core Java is needed.
+- Introductory path of the servlet and its life cycle is needed.
+- Some basic knowledge of HTML is also required.
+You will need to install the following tools to go along with the article:
+- An open-source editor, such as Eclipse, or any other related Java editor.
+- Java JDK current version or any.
+- You need a web browser installed on your computer.
+- Server. In my case, I will be using TomCat version 10.1.0-m5
+
+
+Table of contents:
+
+- [ServletRequest Interface](ServletRequest-Interface )	
+- [Methods of ServletRequest interface](#Methods-of-ServletRequest-interface)	
+- [Example of ServletRequest interface](#Example-of-ServletRequest-interface)	 
+- [RequestDispatcher Interface](#RequestDispatcher-Interface)	
+- [Methods of RequestDispatcher interface](#Methods-of-RequestDispatcher-interface)	
+- [Example of RequestDispatcher](#Example-of-RequestDispatcher)
+- [Conclusion](#conclusion)
+### ServletRequest Interface.
+When we talk of websites, we do interact with,
+ we keep asking web pages to do something for us, for example, client updating their profiles, etc. Whenever a client or user tries to update their profile. In this case. Java provides us with an object of the ServletRequest interface, which forwards client requested information to a servlet. 
+The service container creates ServletRequest and ServletResponse objects when a client sends a request to a web server, which passe as an argument in the service method.
+ The request object allows access to request information such as the header and the requested data body information.
+
+
+### Methods of ServletRequest interface
+The ServletRequest interface provides us with many methods to work with client request information. Some explained as follows.
+
+- `String getParameter(String variableName)`: This method returns a value of parameter requested by a client of a return type `String`. 
+
+- `String[] getParameterValues(type parameterName)`: Majorly used to obtained all values of a given multi-valued parameter name. In this case, useful while working with the multi-select list box where a single parameter may contain multiple values. 
+
+- `public int getContentLength()`: It  returns the actual size of requested data of type `int`.
+
+- `Public String getContentType()`:  Returns the data of a request Internet Media Type, or null, if unknown. 
+
+- `Enumeration getParameterNames()`: It’s used to return an enumeration of all target parameter name.
+
+- `String getCharacterEncoding()`: This method returns the name if the encoded character is specified. If not specified, it returns null in the body of the request.
+
+- `public int getServerPort()`: This method returns the port number that received this request.
+
+- ``public abstract String getServerName()``:  It returns the server name which receives the request.
+
+- `ServletInputStream getInputStream(): throws IOException`: Returns the binary data input stream in the query body. 

```

Comment:
throws  IOException is not part of the function. Please exclude it in this case.